### PR TITLE
Remove errant `__init__.py` that prevents mypy from working

### PR DESCRIPTION
**Pull request by ml-evs**
The file at `src/__init__.py` is not necessary and is somehow breaking mypy for the rest of the project. This PR removes that file, then adds a series of typing related fixes.

### CODE REVIEW

**Review by davidwaroquiers — APPROVED** [review]

**Review by gpetretto — APPROVED** [review]

**Comment by ml-evs** [comment]
@gpetretto I had a go at fixing some of the typing issues suggested by mypy but I'm not entirely convinced by a lot of the annotations I am adding. Would you be able to take a look? (probably before merging this?)

**Comment by gpetretto** [comment]
Sure. Do you want me to try to apply this change locally and check the typing issues? Or do you already have a list of problematic ones to be verified?

**Comment by ml-evs** [comment]
> Sure. Do you want me to try to apply this change locally and check the typing issues? Or do you already have a list of problematic ones to be verified?

I'll push the changes I'm happy with to this PR and try to point out the ones that are weird

**Comment by ml-evs** [comment]
> > Sure. Do you want me to try to apply this change locally and check the typing issues? Or do you already have a list of problematic ones to be verified?
> 
> I'll push the changes I'm happy with to this PR and try to point out the ones that are weird

I've just pushed these, the remaining lint errors are the weird ones (mostly lots of awkward typing around the db potentially returning `None`) -- hopefully the enum value changes are fine though

**Comment by gpetretto** [comment]
I checked some of the errors related to the `None` and, while they are not real errors as the check on the object not being `None` is performed somewhere else, mypy is likely not able to deduce that. Not sure what is the best strategy here: either skip the check or adding some redundant check on the object not being None. I can do that, so I will also check that they are all indeed correct.

The other errors could probably be resolved by adding `__all__` to the `__init__.py` files that have imports.

I am not sure about replacing the values of the Enums with the Enums instance in the CLI. Does it work? Should I test it?

**Comment by ml-evs** [comment]
I've finally got around to fixing the mypy issues in [b7b7f81](https://github.com/Matgenix/jobflow-remote/pull/54/commits/b7b7f8116ac4d0f3783020d6f0fba34593e6e83b), mostly by just adding checks for `None` where appropriate. Could you take another look @gpetretto?

File: src/jobflow_remote/cli/job.py
Comment (gpetretto) [inline src/jobflow_remote/cli/job.py:712]: In the call to `execute_multi_jobs_cmd` below it keeps passing `resources_value`, so `resources` is never used.

File: src/jobflow_remote/cli/job.py
Comment (gpetretto) [inline src/jobflow_remote/cli/job.py]: This breaks the CLI. I always get an error 
> RuntimeError: Type not yet supported: <class 'jobflow_remote.cli.types.DictType'>

that goes away if remove this line and get back to `Optional` instead of `None`

File: src/jobflow_remote/cli/job.py
Comment (gpetretto) [inline src/jobflow_remote/cli/job.py]: Switching to the Enum also breaks the CLI. I think I had tried at the time. Typer expects a string (as it would from the user, and then converts it to the value. Can we revert back these to be the Enum values? Or does it work for you?

File: src/jobflow_remote/cli/project.py
Comment (gpetretto) [inline src/jobflow_remote/cli/project.py]: Enum value as above

File: src/jobflow_remote/cli/runner.py
Comment (gpetretto) [inline src/jobflow_remote/cli/runner.py]: Enum value as above

File: src/jobflow_remote/cli/runner.py
Comment (gpetretto) [inline src/jobflow_remote/cli/runner.py]: Enum value as above

**Review by gpetretto — COMMENTED** [review]
I am fine with most of the changes, but some of those to the CLI seem to break Typer. Does it work for you? Otherwise I would revert them and ignore any related linter comments.

File: src/jobflow_remote/cli/job.py
Comment (ml-evs) [reply src/jobflow_remote/cli/job.py]: Hmmm, that's annoying, as this change is made automatically by pyupgrade... I'll investigate further

File: src/jobflow_remote/cli/job.py
Comment (ml-evs) [reply src/jobflow_remote/cli/job.py]: I think I've caught and reverted all of these, just had to add various `# type: ignore`s

File: src/jobflow_remote/cli/job.py
Comment (ml-evs) [reply src/jobflow_remote/cli/job.py]: Here are the related issues: https://github.com/tiangolo/typer/issues/533 and https://github.com/tiangolo/typer/issues/348 -- for now we can disable this kind of rewrite in pyupgrade (I will try to go through and check that there are no other occurences)

File: src/jobflow_remote/cli/job.py
Comment (ml-evs) [reply src/jobflow_remote/cli/job.py:712]: I have updated this but cannot easily test

**Comment by ml-evs** [comment]
Annoyingly this is now back in the state where the tests work locally but not in the CI; I'll try to debug it when I get a chance but for now do not let this PR hold up the other ones that actually add features.

**Comment by gpetretto** [comment]
> Annoyingly this is now back in the state where the tests work locally but not in the CI; I'll try to debug it when I get a chance but for now do not let this PR hold up the other ones that actually add features.

After your last commit it seems that only a lint issue remains in the tests. Depending on your availability, I suppose these can be fixed and then we can merge this before the other PRs.

**Comment by ml-evs** [comment]
> After your last commit it seems that only a lint issue remains in the tests. Depending on your availability, I suppose these can be fixed and then we can merge this before the other PRs.

Think that's got it -- I won't be able to take a look at the other PRs again until Friday so feel free to do with this as you please!

**Pull request merged**
